includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -630,6 +630,39 @@ public function scripts() {
 		}
 	}
 
+	/**
+	 * Returns true if given product needs shipping. For bundled products it will return true if bundle is not virtual (assembled)
+	 * otherwise if it's virtual bundle it will check if at least one bundled item needs shipping in order to return true.
+	 *
+	 * @since   x.x.x
+	 * @version x.x.x
+	 * @return  boolean True if product needs shipping.
+	 */
+	private function product_needs_shipping( $product ) {
+		if ( ! wc_shipping_enabled() ) {
+			return false;
+		}
+
+		$needs_shipping = $product->needs_shipping();
+
+		if ( class_exists( 'WC_Product_Bundle' ) && is_a( $product, 'WC_Product_Bundle' ) ) {
+
+			// If bundle is unassembled we need to check if at least one item needs shipping
+			if ( $product->is_virtual() ) {
+				$bundled_items = $product->get_bundled_items();
+
+				foreach ( $bundled_items as $bundled_item ) {
+					if ( $bundled_item->get_product()->needs_shipping() ) {
+						$needs_shipping = true;
+						break;
+					}
+				}
```

Comment:
> Honestly I think it's a flaw from the product bundles plugin to not return true for $product->needs_shipping() when it's supposed to. 

I agree with you here.

> This could become an issue to be worked on for the product bundles developer.

I pinged Chris (from SomewhereWarm, plugin author) to look at this PR. I got not response yet. I'll ping them again so they can maybe update their plugin to handle this on their side so we can remove this custom code in the future. Or at least leave comment explaining their decision of why `$product->needs_shipping();` doesn't return the correct value for unassembled bundles.

> Ultimately, I believe a better short-term fix for this would be refactoring the front-end to update the payment request object when requestShipping has changed.

Yeah, we need to do that in a separate issue. I think it will take considerable effort and could change many of the changes I did as part of #1563.

Reply:
Unless there's something else you want me to review, I think we can move this PR to `status: blocked` until we hear back from Chris (the developer).

Reply:
@ricardo I think there's no need to delay this PR. Since this PR will not be merged to `trunk` I can still address any other input on #1610 or in #1602 directly.

Reply:
Hey folks, a couple notes:

- Unassembled bundles are just "Virtual" products meant to manage the cart lifecycle of the products they contain.
- Assembled bundles are a lot more complex, but you can be certain that when a Bundle is assembled, _something_ needs to be shipped.

With that said, Bundles don't necessarily have a static configuration of products. They may contain optional products, and mixes of Virtual/physical items, so to answer the question "does this bundle need shipping?" we need to know "what did the user select?". In cart context, you'll want to be looking at shipping package data to know if there's anything that needs shipping.

Reply:
I've tested Bundles with https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1628 and all this logic won't be necessary 🎉